awscrt/auth.py

```diff
@@ -255,6 +255,30 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_py_provider(cls, py_provider):
+        """
+        Creates a provider that sources credentials from a python class.
+
+        The python provider needs to have a method :get_credential(), which returns
+        the credential as a dictionary like (Expiration and SessionToken are optional):
+        {
+            "AccessKeyId": "accesskey",
+            "SecretAccessKey": "secretAccessKey",
+            "SessionToken": "....",
+            "Expiration": 1247169778
```

Comment:
Add a note that explains expiration's time properties (units, base line, etc...)